### ISSUE TITLE
Questions & inline todos removal

### ISSUE THREAD

**Issue by oraclown** [tellor-io/tellor-parachain-contracts#35]
1. Should the `Parachain not registered` require statement in `reportStakeDeposited` be removed, as the function is internal?
2. Should the `Parachain not registered` require statement in `reportStakeWithdrawRequested` be removed, as the function is internal?
3. Should the `Parachain not registered` require statement in `reportSlash` be removed, as the function is internal?
4. Should the `Parachain not registered` require statement in `reportStakeWithdrawn` be removed, as the function is internal?
5. Should the `Parachain not registered` require statement in `reportVoteExecuted` be removed, as the function is internal?
6. Should the staking contract arg in the `init` function of `ParachainGovernance.sol` be address payable?
7. Should both token transfers in `executeVote` be wrapped in require statements?
8. When `_parachainStakeInfo._account` is set in `depositParachainStake`, should we check if the oracle consumer chain account is already linked to another staker?
9. ..

**Comment by oraclown** [comment]
@tkernell Thoughts?

**Comment by tkernell** [comment]
> When _parachainStakeInfo._account is set in depositParachainStake, should we check if the oracle consumer chain account is already linked to another staker?

Yes we probably should. The only downside is that anyone can claim to own a given "account" and prevent the real account owner from using it. But I wouldn't imagine that someone absolutely has to report from a specific account.

**Comment by tkernell** [comment]
> Should we check if stake info exists before editing any state in requestParachainStakeWithdraw?

I believe that the existing 2 checks should be sufficient, especially the "stakeBalance > amount" check
